### DISCUSSION TITLE
What is the proper way to pass `run_config` to the `ScheduleDefinition` class?

### DISCUSSION THREAD

**Discussion by bennieregenold** [dagster-io/dagster#17553] [Q&A] [answered]
[In the docs](https://docs.dagster.io/_apidocs/schedules-sensors#dagster.ScheduleDefinition) it says to pass in `run_config` as a mapping. I've tried multiple different things, but none of them properly pass the config when I materialize the asset.

Here is my basic setup:
```python
# class to hold the config
class AWSdbtConfig(Config):

    dbt_run_command: list 

# asset that uses the config
@asset(
    group_name='dbt',
)
def dbt_run(aws_api_resource: AWSAPIResource, config: AWSdbtConfig):

    ...

    overrides = { 
        "containerOverrides": [
            {
                "name": "dbt-runner", 
                "command": config.dbt_run_command # this is what actually passes in the run command
            }
        ]
    }
   ...

# schedule that runs the asset
dbt_assets = AssetSelection.groups('dbt')

dbt_work_hours = ScheduleDefinition(
    job=define_asset_job(
        name='dbt_work_hours', 
        selection=dbt_assets,
        tags={
            'alert_tag': 'dbt_prod',
            'dagster/isolation': 'disabled'
        }
    ),
    run_config={
        'ops': {
            'dbt_run': {
                'config': {
                    'dbt_run_command': ["dbt build"]
                }
            }
        }
    },
    cron_schedule="5/30 11-22 * * *" # Every 30 minutes at 5 past from 7am - 6pm ET
)

```

This setup is no passing the run config through. I've also tried it like the following which didn't work either:

```python
run_config=RunConfig(
        ops={"dbt_run": AWSdbtConfig(dbt_run_command=["dbt build --exclude tag:legacy package:dbt_artifacts"])}
    )
```

When I go into the UI it shows this when I try and materialize it:
`
Missing required config entry "ops" at the root. Sample config for missing entry: {'ops': {'dbt_run': {'config': {'dbt_run_command': []}}}}
`

**Comment by sryza** [comment, marked as answer]
Hi @bennieregenold - if you provide run config to the `ScheduleDefinition`, then the config will only apply to runs that are launched by that schedule. If you go to the UI and launch the job, then run config on the schedule won't apply to that run, because that run is launched by you, not the schedule.

If you want the config to apply both to runs that are launched by the schedule and runs that you launch manually in the UI, you can set the run config on the job itself using the `config` argument on `define_asset_job`.

**Reply by bennieregenold** [reply]
that worked, thank you!!
